### PR TITLE
Add script to make sure CHANGELOG is always in sync with `__version__`

### PR DESCRIPTION
**Pull request by Kludex**
Inspired by yesterday's https://github.com/encode/starlette/pull/1713.

As proposed by @tomchristie on the last comment there, this script only makes sure the first version string found on `release-notes.md` matches with the first version string found on `__init__.py`.

If this is accepted, I'll create PRs on the other encode projects.

### CODE REVIEW

**Comment by adriangb** [comment]
This seems reasonable to me. It codifies a currently manual step.

File: scripts/lint [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 export PREFIX=""
-if [ -d 'venv' ] ; then
+if [ -d 'venv' ]; then
     export PREFIX="venv/bin/"
 fi
 export SOURCE_FILES="starlette tests"
 
 set -x
 
+./scripts/sync-version
```
Comment (florimondmanca) [inline scripts/lint]: Should this be in `scripts/check` instead? `scripts/lint` doesn't get run by CI, but check does (mypy, flake8, black checks, etc).

File: scripts/lint [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh -e
 
 export PREFIX=""
-if [ -d 'venv' ] ; then
+if [ -d 'venv' ]; then
```
Comment (Kludex) [inline scripts/lint]:
```suggestion
if [ -d 'venv' ] ; then
```

**Review by euri10 — APPROVED** [review]

**Comment by Kludex** [comment]
I'm going to wait for:
- https://github.com/encode/httpx/pull/2297/
- https://github.com/encode/uvicorn/pull/1560/

Once all are approved, I'm going to merge them all.

**Comment by Kludex** [comment]
Merging this as the analogous was merged on `uvicorn`.

**Pull request merged**